src/Compilers/Core/Portable/InternalUtilities/AdditionalTextComparer.cs

```diff
@@ -53,7 +53,7 @@ public bool Equals(AdditionalText? x, AdditionalText? y)
         public int GetHashCode(AdditionalText obj)
         {
             return Hash.Combine(PathUtilities.Comparer.GetHashCode(obj.Path),
-                                ByteSequenceComparer.GetHashCode(obj.GetText()?.GetChecksum() ?? ImmutableArray<byte>.Empty));
+                                ByteSequenceComparer.GetHashCode(GetTextOrNullIfBinary(obj)?.GetChecksum() ?? ImmutableArray<byte>.Empty));
```

Comment:
Could we add a test for this issue (and the previous issue if not already covered)?